syntax = "proto3";

package yandex.mdb.mlock.api.v1;

import "google/protobuf/timestamp.proto";

option go_package = "a.yandex-team.ru/cloud/mdb/mlock/api;mlock";

// Mlock conflict.
message Conflict {
  string object = 1;

  repeated string ids = 2;
}

// Mlock lock status.
message LockStatus {
  string id = 1;

  string holder = 2;

  string reason = 3;

  google.protobuf.Timestamp create_ts = 4;

  bool acquired = 5;

  repeated string objects = 6;

  repeated Conflict conflicts = 7;
}

// Mlock lock.
message Lock {
  string id = 1;

  string holder = 2;

  string reason = 3;

  google.protobuf.Timestamp create_ts = 4;

  repeated string objects = 5;
}
